server/bin/server.dart

```diff
@@ -3,32 +3,49 @@ import 'dart:io';
 import 'package:args/args.dart';
 import 'package:shelf/shelf.dart' as shelf;
 import 'package:shelf/shelf_io.dart' as io;
+import 'package:shelf_router/shelf_router.dart';
 
-// For Google Cloud Run, set _hostname to '0.0.0.0'.
+import 'endpoints/endpoints_registry.dart';
+import 'endpoints/four_oh_four.dart';
+
+const _localhost = 'localhost';
 const _hostname = '0.0.0.0';
 
 void main(List<String> args) async {
-  var parser = ArgParser()..addOption('port', abbr: 'p');
+  var parser = ArgParser()
+    ..addOption('port', abbr: 'p')
+    ..addFlag(_localhost, defaultsTo: false);
   var result = parser.parse(args);
 
-  // For Google Cloud Run, we respect the PORT environment variable
   var portStr = result['port'] ?? Platform.environment['PORT'] ?? '8080';
   var port = int.tryParse(portStr);
 
+  var host = _hostname;
+  if (result[_localhost]) {
+    host = _localhost;
+  }
+
   if (port == null) {
     stdout.writeln('Could not parse port value "$portStr" into a number.');
     // 64: command line usage error
     exitCode = 64;
     return;
   }
 
+  final api = EndpointsRegistry();
+  final router = Router();
+
+  router.mount('/api/', api.router);
+  router.all('/echo/<ignored|.*>', _echoRequest);
+  router.add404();
```

Comment:
- api/ is handled by api.router
- echo is just to call hello world and test
- and added a custom 404